### ISSUE TITLE
console error in media library

### ISSUE THREAD

**Issue by JulienMelissas** [Interfacelab/ilab-media-tools#10]
I just upgraded our multisite with the new version of the plugin. Thanks for the changes! When I go to click a media library item now I get this error in my console:
```
...
Uncaught TypeError: Cannot read property 'CacheControl' of undefined
...
```

I don't have any of the new AWS fields filled out, including CacheControl. Shouldn't those be optional/backwards compatible?

I'll see if I can push a fix up if the issue isn't fixed yet tomorrow or the next day - I've got a deadline today.

**Comment by jawngee** [comment]
Is there a stack trace at all?

Sent from my iPhone

On Apr 25, 2017, at 10:36 PM, Julien Melissas <notifications@github.com>
wrote:

I just upgraded our multisite with the new version of the plugin. Thanks
for the changes! When I go to click a media library item now I get this
error in my console:

...
Uncaught TypeError: Cannot read property 'CacheControl' of undefined
...

I don't have any of the new AWS fields filled out, including CacheControl.
Shouldn't those be optional/backwards compatible?

I'll see if I can push a fix up if the issue isn't fixed yet tomorrow or
the next day - I've got a deadline today.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/Interfacelab/ilab-media-tools/issues/10>, or mute the
thread

**Comment by JulienMelissas** [comment]
Sorry for not including it in the first place 😕 

```
Uncaught TypeError: Cannot read property 'CacheControl' of undefined
    at eval (eval at m.template (underscore.min.js?ver=1.8.3:5), <anonymous>:124:30)
    at c (underscore.min.js?ver=1.8.3:5)
    at n.template (wp-util.js?ver=4.7.4:34)
    at n.render (media-views.js?ver=4.7.4:2801)
    at n.render (media-grid.js?ver=4.7.4:121)
    at wp.Backbone.Subviews.<anonymous> (wp-backbone.js?ver=4.7.4:302)
    at underscore.min.js?ver=1.8.3:5
    at Function.m.each.m.forEach (underscore.min.js?ver=1.8.3:5)
    at wp.Backbone.Subviews._attach (wp-backbone.js?ver=4.7.4:298)
    at wp.Backbone.Subviews.<anonymous> (wp-backbone.js?ver=4.7.4:210)
(anonymous) @ VM8095:124
c @ underscore.min.js?ver=1.8.3:5
(anonymous) @ wp-util.js?ver=4.7.4:34
render @ media-views.js?ver=4.7.4:2801
render @ media-grid.js?ver=4.7.4:121
(anonymous) @ wp-backbone.js?ver=4.7.4:302
(anonymous) @ underscore.min.js?ver=1.8.3:5
m.each.m.forEach @ underscore.min.js?ver=1.8.3:5
_attach @ wp-backbone.js?ver=4.7.4:298
(anonymous) @ wp-backbone.js?ver=4.7.4:210
(anonymous) @ underscore.min.js?ver=1.8.3:5
m.each.m.forEach @ underscore.min.js?ver=1.8.3:5
render @ wp-backbone.js?ver=4.7.4:209
render @ wp-backbone.js?ver=4.7.4:376
render @ media-views.js?ver=4.7.4:6336
(anonymous) @ wp-backbone.js?ver=4.7.4:302
(anonymous) @ underscore.min.js?ver=1.8.3:5
m.each.m.forEach @ underscore.min.js?ver=1.8.3:5
_attach @ wp-backbone.js?ver=4.7.4:298
(anonymous) @ wp-backbone.js?ver=4.7.4:210
(anonymous) @ underscore.min.js?ver=1.8.3:5
m.each.m.forEach @ underscore.min.js?ver=1.8.3:5
render @ wp-backbone.js?ver=4.7.4:209
render @ wp-backbone.js?ver=4.7.4:376
attach @ media-views.js?ver=4.7.4:6738
open @ media-views.js?ver=4.7.4:6778
createModal @ media-grid.js?ver=4.7.4:436
initialize @ media-grid.js?ver=4.7.4:394
e.View @ backbone.min.js?ver=1.2.3:1
constructor @ wp-backbone.js?ver=4.7.4:349
constructor @ media-views.js?ver=4.7.4:8498
n @ backbone.min.js?ver=1.2.3:1
n @ backbone.min.js?ver=1.2.3:1
n @ backbone.min.js?ver=1.2.3:1
wp.media @ media-models.js?ver=4.7.4:40
openEditAttachmentModal @ media-grid.js?ver=4.7.4:788
_ @ backbone.min.js?ver=1.2.3:1
m @ backbone.min.js?ver=1.2.3:1
f @ backbone.min.js?ver=1.2.3:1
u.trigger @ backbone.min.js?ver=1.2.3:1
StateMachine.(anonymous function) @ media-views.js?ver=4.7.4:1854
toggleSelectionHandler @ media-views.js?ver=4.7.4:2856
dispatch @ jquery.js?ver=1.12.4:3
r.handle @ jquery.js?ver=1.12.4:3
```

**Comment by jawngee** [comment]
Yeah I found it.  Fix incoming.

**Comment by jawngee** [comment]
Fixed in 303744f31e7afa03fc1c2510aae3458cc0294eb3.

Uploading to wordpress.org now.

**Comment by JulienMelissas** [comment]
Thanks!
